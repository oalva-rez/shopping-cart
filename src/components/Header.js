import React from "react";
import search from "../assets/search.png";
import account from "../assets/account.png";
import cart from "../assets/cart.png";

export default function Header() {
  return (
    <header className="header">
      <h1>Sensuro</h1>
      <div className="header--btns">
        <img src={account} alt="account" />
        <img src={search} alt="search" />
        <img src={cart} alt="cart" />
      </div>
    </header>
  );
}
