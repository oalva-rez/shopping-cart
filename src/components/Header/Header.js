import React from "react";
import search from "../../assets/search.png";
import account from "../../assets/account.png";
import cart from "../../assets/cart.png";
import { Link } from "react-router-dom";

export default function Header(props) {
  return (
    <header className="header">
      <Link to="/" data-testid="home-link">
        <h1>Sensuro</h1>
      </Link>
      <div className="header--btns">
        <img src={account} alt="account" />
        <img src={search} alt="search" />
        <Link className="cart-container" to="/cart" data-testid="cart-link">
          <img src={cart} alt="cart" />
          <div className="cart-count">{props.cart.length}</div>
        </Link>
      </div>
    </header>
  );
}
