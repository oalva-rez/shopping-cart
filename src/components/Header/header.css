.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 20px;
  height: 60px;
  background-color: #fff;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
}
.header a {
  text-decoration: none;
  color: black;
}

.header--btns {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 15px;
}

.cart-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 60px;
  height: 60px;
  padding-left: 2px;
  margin-left: 10px;
  background-color: #fff;
  position: relative;
  cursor: pointer;
}

.cart-count {
  position: absolute;
  top: 72%;
  left: 75%;
  transform: translate(-50%, -50%);
  font-size: 14px;
  width: 14px;
  text-align: center;
  font-weight: bold;
  color: #fff;
  background-color: slategray;
  padding: 2px 4px;
  border-radius: 50%;
  text-align: center;
}
