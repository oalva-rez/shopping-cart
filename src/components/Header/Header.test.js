import { render, screen } from "@testing-library/react";
import Header from "./Header";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";

const MockHeader = (props) => {
  return (
    <BrowserRouter>
      <Header cart={props.cart} />
    </BrowserRouter>
  );
};
const cart = [
  {
    item: {
      id: 1,
      name: "Product 1",
    },
    quantity: 1,
  },
  {
    item: {
      id: 2,
      name: "Product 2",
    },
    quantity: 2,
  },
];

describe("Header", () => {
  it("renders", () => {
    render(<MockHeader cart={cart} />);
    expect(screen.getByText("Sensuro")).toBeInTheDocument();
  });

  it('has link to "/cart"', () => {
    render(<MockHeader cart={cart} />);
    expect(screen.getByTestId("cart-link")).toHaveAttribute("href", "/cart");
  });
  it('has link to home page "/"', () => {
    render(<MockHeader cart={cart} />);
    expect(screen.getByTestId("home-link")).toHaveAttribute("href", "/");
  });
  it("has cart count", () => {
    render(<MockHeader cart={cart} />);
    expect(screen.getByText(`${cart.length}`)).toBeInTheDocument();
  });
  it("has cart count when cart is empty", () => {
    render(<MockHeader cart={[]} />);
    expect(screen.getByText("0")).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const { container } = render(<MockHeader cart={cart} />);
    expect(container).toMatchSnapshot();
  });
});
