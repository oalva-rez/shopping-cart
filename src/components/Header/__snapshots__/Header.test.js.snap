// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <a
      data-testid="home-link"
      href="/"
    >
      <h1>
        Sensuro
      </h1>
    </a>
    <div
      class="header--btns"
    >
      <img
        alt="account"
        src="account.png"
      />
      <img
        alt="search"
        src="search.png"
      />
      <a
        class="cart-container"
        data-testid="cart-link"
        href="/cart"
      >
        <img
          alt="cart"
          src="cart.png"
        />
        <div
          class="cart-count"
        >
          2
        </div>
      </a>
    </div>
  </header>
</div>
`;
